docs/api.rst

```diff
@@ -206,6 +206,51 @@ to handle it, which defaults to print a traceback and ignoring the exception.
     errors. To turn a function into a coroutine, they must be ``async def``
     functions.
 
+.. function:: on_automod_rule_create(rule)
+
+    Called when a new auto moderation rule is created.
+
+    :param rule: The rule that is newly created.
+    :type rule: :class:`AutoModerationRule`
+
+    .. versionadded:: 2.0
+
+.. function:: on_automod_rule_update(before, after)
+
+    Called when an auto moderation rule is updated.
+
+    :param before: The updated rule's old info.
+    :type before: :class:`AutoModerationRule`
+    :param after: The updated rule's new info.
+    :type after: :class:`AutoModerationRule`
+
+    .. versionadded:: 2.0
+
+.. function:: on_automod_rule_delete()
+
+    Called when an auto moderation rule is deleted.
+
+    .. versionadded:: 2.0
+
+.. function:: on_automod_action_executed(action)
+
+    Called when an auto moderation action is executed.
+
+    :param action: The action that was executed.
+    :type action: :class:`AutoModerationAction`
+
+    .. versionadded:: 2.0
+
+.. function:: on_raw_automod_action_executed(event):
```

Comment:
Is there actually any difference between this and `on_automod_action_executed`? Normally raw events exist if the non-raw alternative can fail to be called sometimes as far as I'm aware. (I.e. reaction add)

Reply:
yes, raw exists when the normal one relies on cache
like `on_message_delete` contains the cached message as messages are not cached indefinitely
`on_raw_message_delete` then provides the data discord sends back with little change

Reply:
Then this raw event can be deleted.